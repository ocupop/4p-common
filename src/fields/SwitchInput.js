import React from 'react'
import Skeleton from 'react-loading-skeleton'
import { Checkbox } from 'semantic-ui-react'
import useLoading from '../common/hooks/useLoading'
import Hint from './hint'

const SwitchInput = ({
  className, hint, type, label, field, onChange, toggle, form: {
    errors,
    touched,
  },
}) => {
  // deleting field.value, this throws an error with the Checkbox Element.
  delete field.value

  const [loading] = useLoading()

  return (
    <div className={`form-group ${className}`}>
      {loading ? (
        <Skeleton height={60} width={80} />
      ) : (
        <Checkbox
          {...field}
          type={type}
          checked={field.checked}
          label={label}
          onChange={onChange}
          toggle={toggle}
        />
      )}
      {hint && <Hint content={hint} />}
    </div>
  )
}

export default SwitchInput
