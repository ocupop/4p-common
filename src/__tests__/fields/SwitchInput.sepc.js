import React from "react";
import { create } from "react-test-renderer";
import SwitchInput from '../../fields/SwitchInput'

const mockFieldObject = {
  name: 'switchInput',
  onChange: () => {

  },
  value: '',
}

const mockFormObject = {
  dirty: false,
  errors: {},
  touched: false,
}
describe("SwitchInput component", () => {
  test("Matches the snapshot", () => {
    const switchInput = create(<SwitchInput
      className="switchInput"
      hint="this is a hint"
      type="checkbo"
      field={mockFieldObject}
      form={mockFormObject}
    />);
    expect(switchInput.toJSON()).toMatchSnapshot();
  });
});