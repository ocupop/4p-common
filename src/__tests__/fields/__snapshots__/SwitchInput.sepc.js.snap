// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInput component Matches the snapshot 1`] = `
<div
  className="form-group switchInput"
>
  <div
    className="ui fitted checkbox"
    onChange={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <input
      checked={false}
      className="hidden"
      name="switchInput"
      readOnly={true}
      tabIndex={0}
      type="checkbo"
    />
    <label />
  </div>
  <div
    className="pt-1 ml-1 d-inline-block"
  >
    <i
      className="ri-question-fill text-mid"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </div>
</div>
`;
